server/store/oci/oci.go

```diff
@@ -333,249 +325,111 @@ func (s *store) Delete(ctx context.Context, ref *corev1.RecordRef) error {
 	if err != nil {
-		return ocispec.Descriptor{}, fmt.Errorf("failed to resolve record manifest for subject: %w", err)
+		return fmt.Errorf("failed to attach signature: %w", err)
 	}
 
-	// Create annotations for the signature manifest
-	annotations := make(map[string]string)
+	logger.Debug("Cosign attach signature completed successfully")
 
-	// Copy signature annotations
-	for k, v := range signature.GetAnnotations() {
-		annotations[k] = v
-	}
+	return nil
+}
 
-	// Add OCI-required signature artifact type annotation
-	annotations[SignatureTypeAnnotation] = SignatureArtifactMediaType
+// constructImageReference builds the OCI image reference for a record CID.
+func (s *store) constructImageReference(recordCID string) string {
+	// Get the registry and repository from the config
+	registry := s.config.RegistryAddress
+	repository := s.config.RepositoryName
 
-	// Create the signature manifest with proper OCI subject field
-	manifestDesc, err := oras.PackManifest(ctx, s.repo, oras.PackManifestVersion1_1, SignatureManifestMediaType,
-		oras.PackManifestOptions{
-			ManifestAnnotations: annotations,
-			Subject:             &recordManifestDesc, // OCI 1.1 subject field
-			Layers: []ocispec.Descriptor{
-				signatureDesc,
-			},
-		},
-	)
-	if err != nil {
-		return ocispec.Descriptor{}, fmt.Errorf("failed to pack signature manifest: %w", err)
-	}
+	// Remove any protocol prefix from registry address for the image reference
+	registry = strings.TrimPrefix(registry, "http://")
+	registry = strings.TrimPrefix(registry, "https://")
 
-	return manifestDesc, nil
+	// Use CID as tag to match the oras.Tag operation in Push method
+	return fmt.Sprintf("%s/%s:%s", registry, repository, recordCID)
 }
 
 // ReferrersLister interface for repositories that support the OCI Referrers API.
 type ReferrersLister interface {
 	Referrers(ctx context.Context, desc ocispec.Descriptor, artifactType string, fn func(referrers []ocispec.Descriptor) error) error
 }
 
-// findSignatureManifest finds the signature manifest for a given record using OCI Referrers API.
-func (s *store) findSignatureManifest(ctx context.Context, recordCID string) (*ocispec.Descriptor, error) {
-	logger.Debug("Finding signature manifest for record", "recordCID", recordCID)
-
-	// First, resolve the record manifest to get its descriptor
-	recordManifestDesc, err := s.repo.Resolve(ctx, recordCID)
-	if err != nil {
-		return nil, fmt.Errorf("failed to resolve record manifest: %w", err)
-	}
-
-	// Check if the repository supports the referrers API
-	referrerLister, ok := s.repo.(ReferrersLister)
-	if !ok {
-		logger.Debug("Repository does not support Referrers API, falling back to tag schema")
-
-		return nil, errors.New("repository does not support Referrers API")
+// VerifyWithZot queries zot's verification API to check if a signature is valid.
+func (s *store) VerifyWithZot(ctx context.Context, recordCID string) (bool, error) {
+	verifyOpts := &zot.VerificationOptions{
```

Comment:
should not be zot-specific from the naming perspective

Reply:
we need it here because in the controller we want to be able to check if this store type supports zot verification